engine/dlib/src/dlib/message.h

```diff
@@ -50,20 +50,9 @@ namespace dmMessage
         }
         /// Socket
         HSocket     m_Socket;
-        /// Lua function reference for callback dispatching.
-        /// The value is ref - LUA_NOREF as LUA_NOREF is defined as -2
-        /// and we have the convention of zero for default values.
-        /// To get the actual ref, do ref + LUA_NOREF
-        /// It's unfortunate that we have to add lua related
-        /// functionality here though.
-        ///
-        /// This is a ref done in the instance local context table
-        /// using dmScript::RefInInstance() so when the corresponding
-        /// script instance dies it will be automatically cleaned up
-        /// if the cleanup callback is not called
-        int         m_FunctionRef;
-
-        int         _padding;
+
+        /// Reserved for sub component path
+        dmhash_t    _reserved;
```

Comment:
From our latest discussions around URL's.
We don't want the struct to grow. And the Lua reference was frankly a very context specific thing.

Note that we keep the size and layout the "same", in order to support old save games.

Note that even though we stored (and loaded) the function reference, it would never really work, since the rest of the code might change Lua/C++, which wouldn't quarantuee that the reference is the same.

I consider it a bug that we didn't zero out the function reference before storing it.